bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/RoomHandler.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.warmup.internal.handler;
+
+import static org.openhab.binding.warmup.internal.WarmupBindingConstants.*;
+
+import java.math.BigDecimal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.warmup.internal.api.MyWarmupApiException;
+import org.openhab.binding.warmup.internal.model.query.LocationDTO;
+import org.openhab.binding.warmup.internal.model.query.QueryResponseDTO;
+import org.openhab.binding.warmup.internal.model.query.RoomDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class RoomHandler extends WarmupThingHandler implements WarmupRefreshListener {
+
+    private final Logger logger = LoggerFactory.getLogger(RoomHandler.class);
+    private RoomConfigurationDTO config;
+
+    public RoomHandler(Thing thing) {
+        super(thing);
+        config = getConfigAs(RoomConfigurationDTO.class);
```

Comment:
The configuration should be read in `initialize()`, since the handler object is re-used when the configuration is updated. Then, `dispose()` followed by `initialize()` will be invoked, without calling the constructor.

Reply:
Thank you, this is resolved, although I hope the @SuppressWarnings("null") I added is ok? https://github.com/openhab/openhab-addons/pull/7998/commits/7bc9ed18881442a51c73f95fe123773f6d04e482#diff-782a12acc9c698d8cf1b524ad394758e

Reply:
Sorry for the delay. Please do a null check instead of suppressing the warning. I'm going to approve the code after the fix. @Hilbrand you could do a second review if you want.

Reply:
resolved in https://github.com/openhab/openhab-addons/pull/8562/commits/7be6f8d8ab878b0c37f5040555ce9e4f9707188c